Scripts/Process/MomentProcessor.cs

```diff
@@ -0,0 +1,156 @@
+        }
+
+        [Tooltip("The moment in which to process the given processes.")]
+        public MomentToProcess momentToProcess = MomentToProcess.OnPreRender;
+        [Tooltip("A collection of Moments to process.")]
+        public MomentProcess[] processes = new MomentProcess[0];
+
+        protected MomentToProcess subscribedMoment;
+
+        protected virtual void OnEnable()
+        {
+            subscribedMoment = MomentToProcess.None;
+            ManageSubscriptions();
+        }
+
+        protected virtual void OnDisable()
+        {
+            UnsubscribeMoment();
+        }
+
+        protected virtual void FixedUpdate()
+        {
+            if (momentToProcess == MomentToProcess.OnFixedUpdate)
+            {
+                Process();
+            }
+        }
+
+        protected virtual void Update()
+        {
+            if (momentToProcess == MomentToProcess.OnUpdate)
+            {
+                Process();
+            }
+        }
+
+        protected virtual void LateUpdate()
+        {
+            if (momentToProcess == MomentToProcess.OnLateUpdate)
+            {
+                Process();
+            }
+            ManageSubscriptions();
+        }
+
+        protected virtual void OnCameraPreRender(Camera givenCamera)
+        {
+            Process();
+        }
+
+        protected virtual void OnCameraPreCull(Camera givenCamera)
+        {
+            Process();
+        }
+
+        /// <summary>
+        /// The ManageSubscriptions method handles subscribing and unsubscribing to the relevant camera events.
+        /// </summary>
+        protected virtual void ManageSubscriptions()
+        {
+            if (subscribedMoment != momentToProcess)
+            {
+                UnsubscribeMoment();
+                SubscribeMoment();
+            }
+        }
+
+        /// <summary>
+        /// The UnsubscribeMoment method handles unsubscribing to the chosen subscribed moment event.
+        /// </summary>
+        protected virtual void UnsubscribeMoment()
+        {
+            switch (subscribedMoment)
+            {
+                case MomentToProcess.OnPreRender:
+                    Camera.onPreRender -= OnCameraPreRender;
+                    subscribedMoment = MomentToProcess.None;
+                    break;
+                case MomentToProcess.OnPreCull:
+                    Camera.onPreCull -= OnCameraPreCull;
+                    subscribedMoment = MomentToProcess.None;
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// The SubscribeMoment method handles subscribing to the chosen moment to process event.
+        /// </summary>
+        protected virtual void SubscribeMoment()
+        {
+            switch (momentToProcess)
+            {
+                case MomentToProcess.OnPreRender:
+                    Camera.onPreRender += OnCameraPreRender;
+                    subscribedMoment = MomentToProcess.OnPreRender;

```

Comment:
Same duplication. I think we should move it out of the `switch` and instead move it to be done after it. The line should then just read `subscribedMoment = momentToProcess;`.

Reply:
👍